### PR TITLE
Flagsmith RUM refactor

### PR DESCRIPTION
**Pull request by dabeeeenster**
### What does this PR do?

Refactor the Flagsmith RUM integration into a new integration tile

### Motivation

I was asked to by Dominic Medina!

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [X] Feature or bugfix has tests
- [X] Git history is clean
- [X] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by drichards-87 — APPROVED** [review]

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]
Thank you for opening this PR! I left some comments to fix before we can move forward. 

Additionally, can you please add another `CODEOWNER`s entry for the new `flagsmith-rum` listing? This is technically a new listing, so we need to add all of the appropriate scaffolding for the listing like we did with the original `flagsmith` PR. You can look at other entries as an example (NOTE: `@DataDog/marketplace-review` should be added to each line that you add)

File: flagsmith-rum/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        client: datadogRum,
+        trackTraits: true,
+    },
+    ...
+})
+```
+
+## Troubleshooting
+
+Need help? See the [Flagsmith documentation][4] or contact [Datadog Support][2].
```
Comment (bgoldberg122) [inline flagsmith-rum/README.md:34]: Please add the url for `#4` at the bottom.

File: flagsmith-rum/assets/service_checks.json
Comment (bgoldberg122) [inline flagsmith-rum/assets/service_checks.json:1]: This file can be deleted since this is a tile-only listing.

File: flagsmith-rum/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Agent Check: Flagsmith
```
Comment (anawishnoff) [inline flagsmith-rum/README.md:1]:
```suggestion
```
We can remove this line as this isn't an agent check :)

File: flagsmith-rum/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+{{< callout url="#" btn_hidden="true" header="Join the Feature Flag Tracking Beta!">}}
```
Comment (anawishnoff) [inline flagsmith-rum/README.md:3]: @meghanlo is this feature still in beta? Do we still want to have this callout here?

File: flagsmith-rum/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+{{< /callout >}}
+
+## Overview
+
+[Flagsmith][1] facilitates feature management across web, mobile, and server side applications.
+
+The Datadog Flagsmith RUM integration enriches your RUM data with your feature flags to provide visibility into performance monitoring and behavioral changes. Determine which users are shown a user experience and if it is negatively affecting the user's performance.
+
+### Feature flag tracking setup
```
Comment (anawishnoff) [inline flagsmith-rum/README.md:14]:
```suggestion
## Setup
```

File: flagsmith-rum/manifest.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Enriches your RUM data with your feature flags from Flagsmith",
+    "title": "Flagsmith",
+    "media": [],
+    "classifier_tags": [
+      "Supported OS::Linux",
+      "Supported OS::macOS",
+      "Supported OS::Windows",
+      "Category::Notification"
```
Comment (anawishnoff) [inline flagsmith-rum/manifest.json:18]: A few more recommended categories to add: Issue Tracking, Developer Tools

Additionally, please add the offering type:
`"Offering Type::Integration"`

File: flagsmith-rum/manifest.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      "metrics": {
+        "metadata_path": "metadata.csv",
+        "check": [],
+        "prefix": "flagsmith."
+      },
+      "service_checks": {
+        "metadata_path": "assets/service_checks.json"
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (anawishnoff) [reply flagsmith-rum/manifest.json:44]: +1 on the above - since there is no code that is attached to this tile, this section can be removed.

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few comments from my end to get things standardized

File: flagsmith-rum/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# CHANGELOG - Flagsmith
+
+## 1.0.0
```
Comment (bgoldberg122) [inline flagsmith-rum/CHANGELOG.md:3]:
```suggestion
## 1.0.0 / 2023-5-15
```

File: flagsmith-rum/manifest.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    "classifier_tags": [
+      "Supported OS::Linux",
+      "Supported OS::macOS",
+      "Supported OS::Windows",
+      "Category::Notification"
+    ]
+  },
+  "author": {
+    "support_email": "support@flagsmith.com",
+    "homepage": "https://github.com/DataDog/integrations-extras",
```
[the remark this answers is not in the document]
Comment (bgoldberg122) [reply flagsmith-rum/manifest.json]: Thanks for updating! If you could also update the other `flagsmith` listing's homepage field that would be great.

File: flagsmith/manifest.json [hunk 9]
```diff
@@ -12,12 +12,12 @@
...
-      "Category::Configuration & Deployment",
-      "Category::Notification",
-      "Category::Testing",
       "Supported OS::Linux",
+      "Supported OS::macOS",
       "Supported OS::Windows",
-      "Supported OS::macOS"
+      "Category::Issue Tracking,",
+      "Category::Developer Tools",
+      "Offering Type::Integration"
```
[the remark this answers is not in the document]
Comment (bgoldberg122) [reply flagsmith/manifest.json:20]: Apologies for the confusion, the correct tag is actually `Offering::Integration` (It also looks like an extra `,` snuck into the Category tag)

File: flagsmith-rum/manifest.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Enriches your RUM data with your feature flags from Flagsmith",
+    "title": "Flagsmith",
+    "media": [],
+    "classifier_tags": [
+      "Supported OS::Linux",
+      "Supported OS::macOS",
+      "Supported OS::Windows",
+      "Category::Notification"
```
Comment (anawishnoff) [inline flagsmith-rum/manifest.json:18]:
```suggestion
      "Category::Notification",
      "Category::Issue Tracking",
      "Category::Developer Tools",
      "Offering::Integration"
```

File: flagsmith-rum/manifest.json [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+  "app_id": "flagsmith",
+  "display_on_public_website": true,
+  "tile": {
+    "overview": "README.md#Overview",
+    "configuration": "README.md#Setup",
+    "support": "README.md#Support",
+    "changelog": "CHANGELOG.md",
+    "description": "Enriches your RUM data with your feature flags from Flagsmith",
+    "title": "Flagsmith",
+    "media": [],
```
[the remark this answers is not in the document]
Comment (anawishnoff) [reply flagsmith-rum/manifest.json:13]: +1 on this comment - we'll need an image or two for the RUM integration tile.

File: flagsmith/manifest.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (anawishnoff) [reply flagsmith/manifest.json:20]: Let's also add "Offering::UI Extension" here!

**Review by anawishnoff — CHANGES_REQUESTED** [review]
A few more comments from me. Thanks!

File: flagsmith/manifest.json [hunk 12]
```diff
@@ -12,17 +12,18 @@
...
-      "Category::Notification",
-      "Category::Testing",
       "Supported OS::Linux",
+      "Supported OS::macOS",
       "Supported OS::Windows",
-      "Supported OS::macOS"
+      "Category::Issue Tracking",
+      "Category::Developer Tools",
+      "Offering::UI Extension",
+      "Offering Type::Integration"
```
Comment (anawishnoff) [inline flagsmith/manifest.json:21]:
```suggestion
      "Offering::Integration"
```
Apologies for the confusion here, but this is the correct format

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Still a few unresolved comments from the last review and one new one. Let us know if there are any questions. Thanks!

File: flagsmith-rum/README.md [hunk 13]
```diff
@@ -0,0 +1,36 @@
+{{< callout url="#" btn_hidden="true" header="Join the Feature Flag Tracking Beta!">}}
+To enrich your RUM data with your Flagsmith feature flags and get visibility into performance monitoring and behavioral changes, join the <a href="https://docs.datadoghq.com/real_user_monitoring/guide/setup-feature-flag-data-collection/">Feature Flag Tracking</a> private beta. To request access, contact Datadog Support at support@datadoghq.com.
+
+{{< /callout >}}
```
Comment (anawishnoff) [inline flagsmith-rum/README.md:4]:
```suggestion
```
This should be removed as the feature is no longer in private beta

File: flagsmith-rum/README.md (hunk 3 above)
Comment (meghanlo) [reply flagsmith-rum/README.md:1]: It is still in beta but I'm fine to remove if you prefer

**Comment by dabeeeenster** [comment]
OK two images of RUM added! Think this is all comments dealt with now?

File: flagsmith-rum/README.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
+# Agent Check: Flagsmith / Datadog RUM
+
+## Overview
+
+[Flagsmith][1] facilitates feature management across web, mobile, and server side applications.
```
Comment (bgoldberg122) [inline flagsmith-rum/README.md:5]: Is this meant to be the Flagsmith homepage?

Comment (dabeeeenster) [reply flagsmith-rum/README.md:5]: OK all these are addressed - thanks!

File: flagsmith-rum/manifest.json [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+        "caption": "Datadog RUM with Flagsmith Flags details",
+        "image_url": "images/flag_rum_details.png",
+        "media_type": "image"
+      }
+    ],
+    "classifier_tags": [
+      "Supported OS::Linux",
+      "Supported OS::macOS",
+      "Supported OS::Windows",
+      "Category::Notification",
```
Comment (meghanlo) [inline flagsmith-rum/manifest.json:29]:
```suggestion
      "Category::Configuration & Deployment",
```

nit: perhaps this category makes more sense here

**Review by meghanlo — APPROVED** [review]

File: flagsmith-rum/README.md [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+        client: datadogRum,
+        trackTraits: true,
+    },
+    ...
+})
+```
+
+## Troubleshooting
+
+Need help? See the [Flagsmith documentation][4] or contact [Datadog Support][3].
```
Comment (bgoldberg122) [inline flagsmith-rum/README.md:29]: I think these urls might be swapped, should this be the following?
```suggestion
Need help? See the [Flagsmith documentation][3] or contact [Datadog Support][4].
```

File: .github/CODEOWNERS [hunk 17]
```diff
@@ -57,9 +57,10 @@
 /fiddler/                                @dannybrock sales@fiddler.ai @DataDog/ecosystems-review
 /filebeat/                               @wk8 jean@tripping.com
 /filemage/                               @devopsec
-/finout/                                 @roiravhon support@finout.io @DataDog/ecosystems-review
```
Comment (yzhan289) [inline .github/CODEOWNERS:60]: Can you rebase with master? `marketplace-review` recently changed to `ecosystems-review` so the CODEOWNERS file is out of sync.

**Review by yzhan289 — COMMENTED** [review]
Just 1 note regarding CODEOWNERS file. Also, not sure how strict this requirement is, but I think you may need to rename your `flagsmith-rum` directory to `flagsmith_rum`.

File: .github/CODEOWNERS (hunk 17 above)
Comment (dabeeeenster) [reply .github/CODEOWNERS:60]: OK that's done

**Review by bgoldberg122 — APPROVED** [review]

**Review by yzhan289 — APPROVED** [review]

**Pull request merged**